networkx/readwrite/json_graph/jit.py

```diff
@@ -0,0 +1,99 @@
+"""
+Read and write NetworkX graphs as JavaScript InfoVis Toolkit (JIT) format JSON.
+
+See the JIT documentation and examples at http://thejit.org
+
+Format
+------
+var json = [
+  {
+    "id": "aUniqueIdentifier",
+    "name": "usually a nodes name",
+    "data": {
+      "some key": "some value",
+      "some other key": "some other value"
+     },
+    "adjacencies": [
+    {
+      nodeTo:"aNodeId",
+      data: {} //put whatever you want here
+    },
+    'other adjacencies go here...'
+  },
+
+  'other nodes go here...'
+];
+"""
+
+#    Copyright (C) 2011-2013 by
+#    Aric Hagberg <hagberg@lanl.gov>
+#    Dan Schult <dschult@colgate.edu>
+#    Pieter Swart <swart@lanl.gov>
+#    All rights reserved.
+#    BSD license.
+
+__all__ = ['jit_graph', 'jit_data']
+
+import networkx as nx
+import json
+from networkx.utils.decorators import not_implemented_for
+
+
+def jit_graph(data):
+    """Read a graph from JIT JSON.
+
+    Parameters
+    ----------
+    data : JSON Graph Object
+
+    Returns
+    -------
+    G : NetworkX Graph
+    """
+    G = nx.Graph()
+    for node in data:
+        G.add_node(node['id'])
+        for adj in node['adjacencies']:
+            G.add_edge(node['id'], adj['nodeTo'], **adj['data'])
+            G.add_node(node['id'], **node['data'])
+    return G
+
+
+@not_implemented_for('multigraph')
+def jit_data(G, indent=None):
+    """Return data in JIT JSON format.
+
+    Parameters
+    ----------
+    G : NetworkX Graph
+
+    indent: optional, default=None
+        If indent is a non-negative integer, then JSON array elements and object
+        members will be pretty-printed with that indent level. An indent level
+        of 0, or negative, will only insert newlines. None (the default) selects
+        the most compact representation.
+
+    Returns
+    -------
+    data: JIT JSON string
+    """
+    json_graph = []
+    for node in G.nodes():
```

Comment:
This could be refactored as `_jit_node(node)`, then this function could be made clearer as

    # TODO This can be parallelized.
    json_graph = [_jit_node(v) for v in G]
    return json.dumps(json_graph, indent=indent)

This makes it clear where most of the work is being done, and indicates more clearly where parallelism can be applied.